### DISCUSSION TITLE
How Do i disable a security feture

### DISCUSSION THREAD

**Discussion by showierdata9978** [aio-libs/aiohttp#7004] [Q&A]
I'm currently dealing with the [discourse](https://www.discourse.org/) API, and i would like to disable a  security feature that i do not care about.  

```
Error: ValueError: Newline or carriage return character detected in HTTP status message or header. This is a potential security issue
```

I don't care that there is a new line char in the status code or headers. I just want to send a post request and gets its result back.

**Comment by Dreamsorcerer** [comment]
Just remove the newline from the headers you're creating?

The reason this is an issue, is that the newline will result in the server reading the part after the newline as a different header. Surely not what you intended...

e.g.
`Authorization: apple\nfoo: bar`
is exactly the same as:
```
Autharization: apple
foo:bar
```
i.e. 2 different headers.

If the value of the header came from an untrusted source, this is a security issue as it allows them to inject their own headers into your requests. If the source is under your control, then your header values are just borked.
